src/watertap_contrib/seto/unit_models/surrogate/med_tvc_surrogate.py

```diff
@@ -0,0 +1,1394 @@
+            )
+            brine_mass_flow_tot = sum(
+                brine.flow_mass_phase_comp["Liq", j] for j in brine.component_list
+            )
+            dist_mass_flow_tot = sum(
+                dist.flow_mass_phase_comp["Liq", j] for j in dist.component_list
+            )
+
+            enth_feed = pyunits.convert(
+                feed.enth_mass_phase["Liq"], to_units=pyunits.kJ / pyunits.kg
+            )
+            enth_brine = pyunits.convert(
+                brine.enth_mass_phase["Liq"], to_units=pyunits.kJ / pyunits.kg
+            )
+            enth_dist = pyunits.convert(
+                dist.enth_mass_phase["Liq"], to_units=pyunits.kJ / pyunits.kg
+            )
+            enth_cool = pyunits.convert(
+                cool.enth_mass_phase["Liq"], to_units=pyunits.kJ / pyunits.kg
+            )
+            return b.feed_cool_mass_flow * (enth_cool - enth_feed) == (
+                (1 - b.thermal_loss) * b.thermal_power_requirement
+                - brine_mass_flow_tot * enth_brine
+                - dist_mass_flow_tot * enth_dist
+                + enth_cool * feed_mass_flow_tot
+            )
+
+        # Volume flow rates
+        @self.Constraint(doc="Feed and cooling water volumetric flow rate (m3/h)")
+        def eq_feed_cool_vol_flow(b):
+            return b.feed_cool_vol_flow == pyunits.convert(
+                b.feed_cool_mass_flow / b.feed_props[0].dens_mass_phase["Liq"],
+                to_units=pyunits.m**3 / pyunits.hour,
+            )
+
+        @self.Constraint(doc="Cooling water volumetric flow rate (m3/h)")
+        def eq_cool_vol_flow(b):
+            return b.feed_cool_vol_flow == pyunits.convert(
+                b.cooling_out_props[0].flow_vol_phase["Liq"]
+                + b.feed_props[0].flow_vol_phase["Liq"],
+                to_units=pyunits.m**3 / pyunits.hr,
+            )
+
+    def initialize_build(
+        blk,
+        state_args=None,
+        outlvl=idaeslog.NOTSET,
+        solver=None,
+        optarg=None,
+    ):
+        """
+        General wrapper for initialization routines
+
+        Keyword Arguments:
+            state_args : a dict of arguments to be passed to the property
+                         package(s) to provide an initial state for
+                         initialization (see documentation of the specific
+                         property package) (default = {}).
+            outlvl : sets output level of initialization routine
+            optarg : solver options dictionary object (default=None)
+            solver : str indicating which solver to use during
+                     initialization (default = None)
+
+        Returns: None
+        """
+        init_log = idaeslog.getInitLogger(blk.name, outlvl, tag="unit")
+        solve_log = idaeslog.getSolveLogger(blk.name, outlvl, tag="unit")
+
+        opt = get_solver(solver, optarg)
+        # ---------------------------------------------------------------------
```

Comment:
You should add a check for degrees of freedom = 0 in the initialization routine.

Reply:
Added at line 834 before solving the model.

Reply:
Test

Reply:
Added at line 834 before solving the model.